File: src/main/java/net/frozenblock/configurableeverything/biome_placement/util/BiomePlacementChange.java
package net.frozenblock.configurableeverything.biome_placement.util;

import com.mojang.serialization.Codec;
import com.mojang.serialization.codecs.RecordCodecBuilder;
import java.util.List;

public record BiomePlacementChange(List<DimensionBiomeList> addedBiomes, List<DimensionBiomeKeyList> removedBiomes) {

	public static final Codec<BiomePlacementChange> CODEC = RecordCodecBuilder.create(instance ->
		instance.group(
			DimensionBiomeList.CODEC.listOf().fieldOf("addedBiomes").forGetter(BiomePlacementChange::addedBiomes),
			DimensionBiomeKeyList.CODEC.listOf().fieldOf("removedBiomes").forGetter(BiomePlacementChange::removedBiomes)
		).apply(instance, BiomePlacementChange::new)
	);
}

File: src/main/java/net/frozenblock/configurableeverything/biome_placement/util/BiomePlacementChange.kt
package net.frozenblock.configurableeverything.biome_placement.util

import com.mojang.serialization.Codec
import com.mojang.serialization.codecs.RecordCodecBuilder
import net.frozenblock.configurableeverything.util.mutListOf

data class BiomePlacementChange(
	@JvmField var addedBiomes: MutableList<DimensionBiomeList>,
	@JvmField var removedBiomes: MutableList<DimensionBiomeKeyList>
) {
	companion object {
        @JvmField
		val CODEC: Codec<BiomePlacementChange> = RecordCodecBuilder.create { instance ->
			instance.group(
				DimensionBiomeList.CODEC.mutListOf().fieldOf("addedBiomes").forGetter(BiomePlacementChange::addedBiomes),
				DimensionBiomeKeyList.CODEC.mutListOf().fieldOf("removedBiomes").forGetter(BiomePlacementChange::removedBiomes)
			).apply(instance, ::BiomePlacementChange)
		}
	}
}
